eosc_pl/transform/utils/loader.py

```diff
@@ -2,37 +2,55 @@
 """Load data"""
 import logging
 import urllib.parse
+from typing import Optional
 
 import pandas as pd
 import requests
-
-from eosc_pl.transform.utils.validate import validate_loaded_pd_df
+from transform.utils.consts import Repository
+from transform.utils.validate import validate_loaded_pd_df
 
 logger = logging.getLogger(__name__)
 
 
-def pd_load_datasets(url: str) -> pd.DataFrame | None:
+def _get_params(per_page: int, start: int, repository: Repository) -> str:
+    repo_to_params_map = {
+        Repository.REPOD: "?q=*&type=dataset&per_page={per_page}&start={start}",
+        Repository.RODBUK: "?q=*&type=dataset&per_page={per_page}&metadata_fields=citation:*",
+    }
+    params_string = repo_to_params_map[repository]
+    return params_string.format(per_page=per_page, start=start)
+
+
+def pd_load_datasets_with_pagination(conf: any) -> pd.DataFrame | None:
     """Load datasets from url as pandas df"""
-    try:
-        response = requests.get(url, timeout=200)
-        response.raise_for_status()  # Raise an exception for any unsuccessful response
-        data = response.json()["data"]["items"]
-        data = pd.DataFrame(data)
-        validate_loaded_pd_df(response, data)
-        return data
-    except requests.exceptions.RequestException as e:
-        logger.error("Error during the request:", e)
-        return None
-    except (KeyError, ValueError) as e:
-        logger.error("Error while parsing the response:", e)
-        return None
-    except Exception as e:
-        logger.error("An unexpected error occurred:", e)
-        return None
+    start = 0
+    total = 10000
+    per_page = conf.PER_PAGE
+    dfs = []
+    while start < total:
+        try:
+            params = _get_params(per_page, start, conf.REPOSITORY)
+            url = f"{conf.DATASET_LIST_ADDRESS}{params}"
+            response = requests.get(url=url, timeout=200)
+            response.raise_for_status()  # Raise an exception for any unsuccessful response
+            data = response.json()["data"]["items"]
+            count_in_response = response.json()["data"]["count_in_response"]
+            data = pd.DataFrame(data, index=range(start, start + count_in_response))
+            total = response.json()["data"]["total_count"]
+            validate_loaded_pd_df(response, data)
+            dfs.append(data)
+        except requests.exceptions.RequestException as e:
+            logger.error("Error during the request:", e)
+        except (KeyError, ValueError) as e:
+            logger.error("Error while parsing the response:", e)
+        except Exception as e:
+            logger.error("An unexpected error occurred:", e)
+        start += per_page
+    return pd.concat(dfs)
```

Comment:
It will raise ValueError if no data was fatched
`return pd.concat(dfs) if dfs else None`